pkg/operator/ceph/pool/radosnamespace/controller.go

```diff
@@ -0,0 +1,312 @@
+	// Fetch the CephBlockPoolRadosNamespace instance
+	cephBlockPoolRadosNamespace := &cephv1.CephBlockPoolRadosNamespace{}
+	err := r.client.Get(r.opManagerContext, request.NamespacedName, cephBlockPoolRadosNamespace)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			logger.Debug("cephBlockPoolRadosNamespace resource not found. Ignoring since object must be deleted.")
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, errors.Wrap(err, "failed to get cephBlockPoolRadosNamespace")
+	}
+
+	// Set a finalizer so we can do cleanup before the object goes away
+	err = opcontroller.AddFinalizerIfNotPresent(r.opManagerContext, r.client, cephBlockPoolRadosNamespace)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "failed to add finalizer")
+	}
+
+	// The CR was just created, initializing status fields
+	if cephBlockPoolRadosNamespace.Status == nil {
+		r.updateStatus(r.client, request.NamespacedName, cephv1.ConditionProgressing)
+	}
+
+	// Make sure a CephCluster is present otherwise do nothing
+	cephCluster, isReadyToReconcile, cephClusterExists, reconcileResponse := opcontroller.IsReadyToReconcile(r.opManagerContext, r.client, request.NamespacedName, controllerName)
+	if !isReadyToReconcile {
+		// This handles the case where the Ceph Cluster is gone and we want to delete that CR
+		// We skip the deleteRadosNamespace() function since everything is gone already
+		//
+		// Also, only remove the finalizer if the CephCluster is gone
+		// If not, we should wait for it to be ready
+		// This handles the case where the operator is not ready to accept Ceph command but the cluster exists
+		if !cephBlockPoolRadosNamespace.GetDeletionTimestamp().IsZero() && !cephClusterExists {
+			// Remove finalizer
+			err = opcontroller.RemoveFinalizer(r.opManagerContext, r.client, cephBlockPoolRadosNamespace)
+			if err != nil {
+				return opcontroller.ImmediateRetryResult, errors.Wrap(err, "failed to remove finalizer")
+			}
+
+			// Return and do not requeue. Successful deletion.
+			return reconcile.Result{}, nil
+		}
+		return reconcileResponse, nil
+	}
+
+	// Populate clusterInfo during each reconcile
+	r.clusterInfo, _, _, err = mon.LoadClusterInfo(r.context, r.opManagerContext, request.NamespacedName.Namespace)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "failed to populate cluster info")
+	}
+	r.clusterInfo.Context = r.opManagerContext
+
+	// DELETE: the CR was deleted
+	if !cephBlockPoolRadosNamespace.GetDeletionTimestamp().IsZero() {
+		logger.Debugf("delete ceph blockpool rados namespace %q", cephBlockPoolRadosNamespace.Name)
+		// On external cluster, we don't delete the rados namespace, it has to be deleted manually
+		if cephCluster.Spec.External.Enable {
+			logger.Warning("external rados namespace deletion is not supported, delete it manually")
```

Comment:
We can create a rados namespace on an external cluster but not delete it? Why is that? Is this documented for users? Since this may be important, is there a unit or CI test that verifies this?

Reply:
We dont create the rados namespace on an exluster cluster. we assume its present.